components/layout/display_list_builder.rs

```diff
@@ -1088,8 +1088,11 @@ impl FragmentDisplayListBuilding for Fragment {
     fn create_stacking_context(&self,
                                base_flow: &BaseFlow,
                                display_list: Box<DisplayList>,
-                               layer: Option<Arc<PaintLayer>>)
+                               layout_context: &LayoutContext,
+                               potential_layer_id: LayerId,
+                               layer: Option<PaintLayer>)
```

Comment:
Looks like `potential_layer_id` is never used unless `layer` is `None`. Could you refactor it to something like this:

    enum LayerForStackingContext {
        Existing(PaintLayer),
        New(LayerId),
    }